### ISSUE TITLE
aRef changed

### ISSUE THREAD

**Issue by niekdemelker** [Tools4everBV/HelloID-Conn-Prov-Target-AFAS-Profit-Users#4]
The aRef used to be a string with username, now this is changed to an object. This breaks when trying to update the script in HelloID to a newer version, and doesn't seem to add anything.

Is it acceptable to store an object as reference? and should we want this?

**Comment by rschouten97** [comment]
Hi Niek,

The aRef first wasn't used correctly in this connector, hence the changes.
The aRef is supposed to be used whenever a lookup is done in any action (apart from the create, since it is created/stored in that action). Using the aRef as the unique identifier for a lookup eliminates the issue in which an action can not be performed when the person would have been removed from HelloID.

E.g. the lookup value is the person externalId ($p.externalId).
A person is no longer available in the GetConnectors and thus deleted in HelloID.
HelloID would perform a delete action in this case, but this action would fail, because the $p.externalId doesn't exist anymore.
However, the aRef still exists in this case, so if we store the lookup value/unique identifier in here, we can still perform the delete action.

You are correct in saying that copy-pasting this to an existing connector would break this.

The use of an object as the reference is certainly acceptable and in the end a better fit than a single value.
Since the aRef contains the unique identifier for lookups in all actions (apart from the create, since it is created/stored in that action).

This does indeed break when copy-pasting to a previous implemented connector, because there the aRef was only set and not actually used.
It was however, only set with a single value and not an object, so the lookup in these new connector scripts will not work, since they use the object for the lookup.

I think it best to update the ReadMe with a warning that this no longer is a copy-paste for existing connectors.
However, this repository is still just a template with our best practice. Since the use of the object is the best practice in this case, I do not want to revert the aRef back to a single value.

This works out of the box for a new connector, but not for updating an existing connector.
If needed, we can help with updating an existing connector, and even put an example in this repository for the existing connector. But new connectors should always be using the object.

**Comment by niekdemelker** [comment]
Hi @rschouten97,

Thank you for the detailed insights, but i still don't understand why storing the "gebruiker" code in an object instead of a string is a better fit, it still is one value you track. I do get why you want to store the unique identifier in the aRef and i totally agree on that one.

I'm just asking because i like to know why some decisions where made, this is one of the first connections where i see an object in the aRef, so i wanted to know why.

**Comment by rschouten97** [comment]
Hi @niekdemelker
Thanks for your quick reply!

I agree, there is no added functionality of an object vs the string in this case.
If we had multiple possible identifiers an object should be the best solution though.
It has been set into an object now because we want to show that it is possible to use as an object and this way it would be prepared, if in future we would like to add another identifier to the object.

So to sum it up; it is a showcase of the possibility to add it as an object and to prepare for future possible changes in which multiple fields might want to be stored in the aRef.

**Comment by niekdemelker** [comment]
Hi @rschouten97,

Okey, so just showcasing. Thanks for the explanation.
